File: msdialog/src/main/java/com/maple/msdialog/BaseDialog.java
package com.maple.msdialog;

import android.app.Dialog;
import android.content.Context;
import android.graphics.Point;
import android.view.View;
import android.view.WindowManager;
import android.widget.FrameLayout;
import android.widget.LinearLayout;


/**
 * @author maple
 * @time 17/4/6
 */
public class BaseDialog {
    public Context mContext;
    public Dialog dialog;
    public View rootView;

    public BaseDialog(Context context) {
        mContext = context;
    }

    // 设置Dialog宽度：相对于屏幕宽度比例
    public BaseDialog setScaleWidth(double scWidth) {
        rootView.setLayoutParams(new FrameLayout.LayoutParams(
                (int) (getScreenWidth() * scWidth),
                LinearLayout.LayoutParams.WRAP_CONTENT));
        return this;
    }

    //------------------------------------- utils --------------------------------------------------

    public int getScreenWidth() {
        return getScreenInfo().x;
    }

    public int getScreenHeight() {
        return getScreenInfo().y;
    }

    public Point getScreenInfo() {
        Point size = new Point();
        WindowManager windowManager = (WindowManager) mContext.getSystemService(Context.WINDOW_SERVICE);
        windowManager.getDefaultDisplay().getSize(size);
        return size;
    }

    /**
     * 根据手机的分辨率从 dp 的单位 转成为 px(像素)
     */
    public int dp2px(float dpValue) {
        float scale = mContext.getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }

}

File: msdialog/src/main/java/com/maple/msdialog/BaseDialog.kt
package com.maple.msdialog

import android.app.Dialog
import android.content.Context
import android.graphics.Point
import android.view.View
import android.view.WindowManager
import android.widget.FrameLayout
import android.widget.LinearLayout

/**
 * @author maple
 * @time 2017/4/6
 */
open class BaseDialog(var mContext: Context) {
    protected var dialog: Dialog = Dialog(mContext)
    protected var rootView: View? = null

    // 设置Dialog宽度：相对于屏幕宽度比例
    open fun setScaleWidth(scWidth: Double) {
        rootView?.layoutParams = FrameLayout.LayoutParams(
                (screenInfo().x * scWidth).toInt(),
                LinearLayout.LayoutParams.WRAP_CONTENT)

    }

    fun setCancelable(cancel: Boolean) {
        dialog.setCancelable(cancel)
    }

    fun setCanceledOnTouchOutside(cancel: Boolean) {
        dialog.setCanceledOnTouchOutside(cancel)
    }

    //------------------------------------- utils --------------------------------------------------
    fun screenInfo(): Point {
        val size = Point()
        val windowManager = mContext.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        windowManager.defaultDisplay.getSize(size)
        return size
    }

    /**
     * 根据手机的分辨率从 dp 的单位 转成为 px(像素)
     */
    fun dp2px(dpValue: Float): Int {
        val scale = mContext.resources.displayMetrics.density
        return (dpValue * scale + 0.5f).toInt()
    }

}